research/README.md

```diff
@@ -0,0 +1,325 @@
+
+Adding a large file is handled, showing an indeterminate progress bar, which helps inform the user that something is happening.
+
+| desktop during-file-drop | desktop post-file-drop |
+|---------------------|----------------------|
+| ![](img/002-add-file-desktop-2.png) | ![](img/002-add-file-desktop-3.png) |
+
+There are 2 buttons in the bottom left of the Files pane to open a native file picker dialog. The `+` opens the dialog configured to allow you to pick a single file. The folder icon opens a dialog configured to allow only directories to be selected. This should be collapsed into a single feature.
+
+**Companion**  similar to webUI provides a demarked file-drop area with a button to launch a native file picker dialog. Companion calls the feature "Share files via IPFS" and shows the connected peers count.
+
+| companion pre-file-drop | companion post-file-drop |
+|---------------------|----------------------|
+| ![](img/002-add-file-companion-1.png) | ![](img/002-add-file-companion-2.png) |
+
+The current tab redirects to the url for the file just added. Adding a large file is handled, as the filepicker is a standard html file input. There is no progress indication; the usual browser page loading indicators give a clue that something is happening.
+
+## Sharing files
+
+**WebUI** offers nothing to help the user share a file in their repo. The file list presents a list of CIDs that the user can copy and paste, and manually edit to point to the gateway.
+
+**Desktop** offers a "Copy link" button when the user hovers over a file. Clicking copies the IPFS gateway address of the file to the users clipboard. No indication is given that the action succeeded. No other controls are provided, to edit or remove files.
+
+<img width='600' src='img/002-add-file-desktop-3.png' />
+
+**Companion** adds extra sharing related links to the menu when the current tab is showing a file from IPFS. "Copy Canonical Address", copies a Nested URI `/ipfs/QmHash` address to the clipboard. "Copy Public Gateway URL", copies the full ipfs.io url for the file. Both indicate that it worked with a browser notification. There is no indication to the user that these extra features exist, they only see them by re-opening the ifps browser action menu.
+
+<img width='888' src='img/002-add-file-companion-3.png' />
+
+## Browsing files
+
+Discussion: https://github.com/ipfs-shipyard/pm-ipfs-gui/issues/9
+
+**WebUI** list files by CID. It's not clear what order they are sorted in. Links are provided to open each file in a browser, or explore IPLD (labelled as DAG) data for the node are provided, along with a "x - remove" link.
+
+**Desktop** lists files by filename, sorted with most recently added at the top. The time the file was added is provided as muted, secondary text. In the 0.3, you can't drill down into folders, though I think this feature is available in the latest dev version.
+
+**Companion** has no file browsing feature.
```

Comment:
May be worth mentioning that MFS support for uploaded  files is planned: https://github.com/ipfs-shipyard/ipfs-companion/issues/415
It provides a wider context and implies that we either will have file browser in extension or will open one from dweb/web-ui.